### PR TITLE
feat: Card slot rendering, on_view one-shot, X-Active-Influencer-Id, state-banner tokens

### PR DESCRIPTION
**Pull request by mridulgupta-oi**
## Summary

Coordinated batch across `@one-impression/sdui-runtime` (minor) and `@one-impression/tokens-creator` (minor) to unblock the Explore listing surface and close a correctness gap on view-impression dispatch. Traceability: brief 264 wiki task 035 (`tasks/035-bff-campaigns-explore-listing.md`).

### sdui-runtime (minor)

- **Card renderer — header / items / footer slots.** `CardSnippetSchema` (sdk-native-sdui ^2.8.0) now exposes `data.header?: Node`, `data.items?: Node[]`, `data.footer?: Node`, and a sibling `config?: { footer_bg_color?: ColorToken }`. `CardRenderer` composes them in the legacy `CardSnippetType1` order (header → items → footer). When `config.footer_bg_color` is set, the footer slot is wrapped in a `Box` with the resolved background color so the footer can carry its own banner stripe inside an otherwise-neutral card body.
- **`SduiNode.on_view` — one-shot per instance.** `handleViewed` previously dispatched `props.on_view` every time `Viewable` fired — scroll-back-up over a feed item re-emitted the impression. The wire contract treats `on_view` as a single-impression signal. Added a per-instance `firedRef` that flips to `true` BEFORE the dispatch and short-circuits subsequent calls. New Node instances (e.g. items appended via `append_items` pagination) get their own ref and fire once when scrolled to. The set-before-dispatch ordering matters and is covered by a regression test.
- **`bff_call` — X-Active-Influencer-Id header.** New `useActiveSocialStore` zustand store (`activeInfluencerId: string \| null`, `setActiveInfluencerId(...)`) exported from the package's public surface. `bff_call` always injects the `X-Active-Influencer-Id` header when the store has a value — applied to every environment, not localhost-gated (unlike `X-Dev-Identity`). When `null`, the header is omitted and the server falls back to its default scoping for the authenticated creator.

### tokens-creator (minor)

Four state-banner color pairs in the `sdui.color.*` namespace and a new `palette.state.*` group:
- `state-neutral-{bg,text}` — info / pending review (warm-grey)
- `state-action-required-{bg,text}` — creator must act (amber)
- `state-success-{bg,text}` — approved / paid (green)
- `state-urgent-{bg,text}` — deadline approaching (orange — intentionally distinct from action-required amber and from `negative` red)

Distinct from `palette.status.*`, which is for inline messaging. Bg + text pairs tuned for AA contrast in both themes; dark theme backgrounds use the accent hue at 18% over the dark surface.

## Verification

- `sdk-native-sdui@2.8.0` resolved at install time (verified `node_modules/@one-impression/sdk-native-sdui/package.json`).
- `npm run build` green across all packages (sdui-runtime ESM + types, tokens-creator 7 artifacts including React Native, all other workspaces).
- `npm test` in sdui-runtime: 146 / 147 (1 pre-existing failure in `node-registry.test.ts` from a react-native esbuild transform issue on `origin/main`, unrelated to this PR).
- `npm test` in tokens-creator: 9 / 9 green (including the new `palette.state.*` snapshot + shape tests).
- New tests added (8 total): 3 in `bff-call.test.ts` for `X-Active-Influencer-Id` (set/null/localhost variants); 3 in `useActiveSocialStore.test.ts` for store default + set + clear; 2 in `SduiNode.onView.test.ts` for the firedRef ordering invariant and the behavioural one-shot pattern; 1 new palette state-group test.

## Test plan

- [ ] Reviewer: cross-check Card composition against legacy `one_club_app/src/appLib/snippets/CardSnippetType1/CardSnippetType1.tsx` (header → items → footer ordering, `footer_bg_color` wrap).
- [ ] Reviewer: confirm `firedRef` is set BEFORE `actionEngine.dispatch(props.on_view)` — re-ordering would defeat the re-entrancy guard.
- [ ] Reviewer: confirm `X-Active-Influencer-Id` is NOT localhost-gated (distinct from `X-Dev-Identity`).
- [ ] Reviewer: confirm new tokens render correctly in both light + dark themes (AA contrast check on `*-bg` / `*-text` pairs).
- [ ] Author: validate on iPhone 15 Pro simulator against gateway-side Explore listing handler once that PR lands.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import { fileURLToPath } from "node:url";
+import { dirname, join } from "node:path";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+test("SduiNode — handleViewed guards on a per-instance firedRef before dispatching", () => {
+  // Lock the source-level invariant: handleViewed must check firedRef BEFORE
+  // calling actionEngine.dispatch / telemetry.emit. The ordering matters —
+  // a guard placed after dispatch would still fire on every viewport entry.
+  const src = readFileSync(
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts]: **FLAG** — [test-conventions] The first test in this file validates the one-shot guard by reading and regex-matching the raw source of SduiNode.tsx at runtime (`readFileSync(join(__dirname, '..', 'SduiNode.tsx'))`). Source-scanning tests are brittle: they break on any whitespace/formatting change, they don't survive a build step that renames or inlines the file, and they test the text of the code rather than its behaviour. The second test (lines 65–101) already demonstrates the correct approach — it exercises the guard pattern directly as a pure function. The first test should be replaced with a behavioural test (e.g. using a React test renderer or a lightweight mock of the Viewable callback) that verifies dispatch is called exactly once when handleViewed fires multiple times, rather than asserting on source-code string positions.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped PR that delivers three additive runtime capabilities and eight new design tokens. Code is correct, tests cover the critical paths (including the set-before-dispatch ordering invariant), and the changeset entries are accurate. One minor source-level test fragility worth flagging, but nothing blocking.**

### What this PR is doing
**Intent:** Add three additive capabilities to sdui-runtime (Card header/footer/items slot rendering, one-shot on_view guard, X-Active-Influencer-Id header injection via a new useActiveSocialStore) and add four state-banner color token pairs to tokens-creator, all to unblock the Explore listing surface (brief 264, task 035).
**Delivered:** The diff fully delivers the stated intent. All three runtime changes are present and tested: Card renderer composes header/items/footer with optional footer_bg_color wrapping (Card.renderer.tsx), SduiNode gains a firedRef one-shot guard with set-before-dispatch ordering (SduiNode.tsx), and bff-call injects X-Active-Influencer-Id unconditionally when the store has a value (bff-call.ts). The new useActiveSocialStore is exported on the public surface (index.ts, state/index.ts). The tokens-creator package adds all eight state-banner tokens (four bg + four text) in both light and dark themes with palette.js, palette.d.ts, and palette.test.js updated consistently. Changesets are present for both packages. No acceptance items from the linked task are visibly missing from the diff.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The set-before-dispatch ordering for firedRef is explicitly tested (lines 55-62 of SduiNode.onView.test.ts) — this is exactly the regression that would be missed by a naive 'dispatch count' test alone, and it's good to see it called out.
- The bff-call tests cover all three cases (prod URL, localhost URL, null store) and correctly clean up store state in finally blocks, preventing test-order pollution.
- The footer_bg_color access pattern (reaching through the original Node rather than the parsed data) is clearly commented and the type cast is explicit — this is the right way to handle the sdk-native-sdui schema gap without widening the parsed type.
- Changeset entries are unusually thorough and accurately describe the backwards-compatibility guarantees, which will help consumers assess upgrade risk.

**Read against:**
- `PR body — brief 264 wiki task 035 (tasks/035-bff-campaigns-explore-listing.md)`
- `PR body — sdui-runtime: Card header/footer slots, on_view one-shot, X-Active-Influencer-Id`
- `PR body — tokens-creator: state-banner color tokens`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts (hunk 1 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts]: **FLAG** — [test-conventions] The first test in this file validates the one-shot guard by reading SduiNode.tsx as a raw string and asserting on source-level patterns (regex match for `useRef(false)`, string index comparison for ordering). This is a source-inspection test, not a behavioural test. It will silently pass even if the guard is broken in a way that doesn't match the regex (e.g. variable renamed to `viewedRef`, or the guard moved into a helper). The second test (lines 65–101) correctly exercises the guard pattern behaviourally and is the stronger of the two. The source-inspection test adds fragility without adding coverage — a refactor that preserves semantics but renames the ref will break the test without breaking the feature. Consider removing the source-inspection test and relying on the behavioural test plus a proper React component test (e.g. via @testing-library/react-native) that mounts SduiNode and calls the Viewable callback twice, asserting dispatch count. If a React renderer is genuinely unavailable in this test suite, the behavioural mirror test (second test) is sufficient on its own.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped PR that delivers three additive runtime features and a new token group. Tests are thorough, changesets are present, and the backwards-compatibility claim holds — no existing behaviour is altered. One minor concern worth the engineer's attention: the source-file-reading test approach for SduiNode is fragile by design, but it's a FLAG not a block.**

### What this PR is doing
**Intent:** Add three additive runtime features to sdui-runtime (Card header/footer slots, on_view one-shot guard, X-Active-Influencer-Id header injection via new useActiveSocialStore) and four state-banner color token pairs to tokens-creator, to unblock the Explore listing surface (brief 264, task 035).
**Delivered:** The diff fully delivers the stated intent. All three runtime changes are present and tested: Card renderer composes header/items/footer with optional footer_bg_color wrapping (Card.renderer.tsx), SduiNode.handleViewed is guarded by firedRef with set-before-dispatch ordering (SduiNode.tsx), and bff-call injects X-Active-Influencer-Id unconditionally when the store has a value (bff-call.ts). The new useActiveSocialStore is exported on the public surface. The tokens-creator package adds all eight state-banner tokens (four bg+text pairs) in both light and dark themes with palette.js, palette.d.ts, and palette.test.js all updated consistently. No linked task acceptance items are available to check against individually, but the PR body's stated deliverables are all present in the diff.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The set-before-dispatch ordering for firedRef is correct and the changeset explicitly calls it out — this is the right way to handle re-entrant synchronous callbacks.
- X-Active-Influencer-Id is intentionally not localhost-gated (unlike X-Dev-Identity), and this distinction is documented in three places: the store JSDoc, the bff-call comment, and the test comment block. Clear and reviewable.
- The footer_bg_color access pattern (casting node to Node & { config?: ... }) is the right approach given that config is a sibling of data on the wire schema, not inside the validated data shape — the comment explaining this is helpful.
- Test cleanup (setActiveInfluencerId(null) in finally blocks) prevents state leakage across test files in the bff-call tests.
- Both light and dark theme token values are present and the changeset explains the AA contrast rationale and why urgent-orange is distinct from action-required amber — good design documentation.

**Read against:**
- `PR body — sdui-runtime three-change description`
- `PR body — tokens-creator four state-banner token pairs`
- `.changeset/runtime-card-onview-active-social.md — feature description`
- `.changeset/tokens-creator-state-banner-colors.md — token rationale`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  // implementation changes, this test still documents what the contract
+  // should look like — a Viewable callback that fires N times must
+  // dispatch the underlying action at most once per Node instance.
+  let firedRef = { current: false };
+  let dispatchCount = 0;
+  const dispatch = (): void => {
+    dispatchCount += 1;
+  };
+
+  const handleViewed = (): void => {
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts:36]: **FLAG** — [test-conventions] The test mirrors the guard logic inline (lines 36-40) rather than importing and exercising the real SduiNode.handleViewed. If the guard is removed or moved inside SduiNode, this test still passes because it is testing its own local copy of the pattern, not the production code. The previous review round flagged this; the diff at this file is unchanged from that round — the inline closure is still the subject under test. To make this a regression guard, either (a) export handleViewed as a testable unit from SduiNode (e.g. via a hook or a standalone helper), or (b) use a React testing library render to mount SduiNode with a mock Viewable and assert dispatch count. The current test documents the intended contract well but would not catch a regression in the actual component.

File: packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+import { test } from "node:test";
+import assert from "node:assert/strict";
+
+test("SduiNode — one-shot guard pattern: repeated calls dispatch exactly once", () => {
+  // Behavioural mirror of the guard inside SduiNode.handleViewed. If the
+  // implementation changes, this test still documents what the contract
+  // should look like — a Viewable callback that fires N times must
+  // dispatch the underlying action at most once per Node instance.
+  let firedRef = { current: false };
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts:30]: **FLAG** — [test-conventions] The test mirrors the one-shot guard by re-implementing it inline (a local `firedRef` + `handleViewed` closure) rather than importing and exercising the real `SduiNode` component. This means the test can pass even if the guard is removed from the actual component — the test validates the pattern in isolation, not the implementation. The previous review round flagged this; the diff at this file is unchanged. To make the test a genuine regression guard, either (a) render `SduiNode` with a test renderer and assert that a mocked `actionEngine.dispatch` is called exactly once across multiple `Viewable` callbacks, or (b) at minimum add a comment acknowledging this is a pattern-documentation test and that the real guard is covered by the bff-call integration tests, so reviewers don't mistake it for a component-level regression test.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped PR that delivers three additive runtime features and four token pairs. Tests are present for all new behaviour. One pre-existing FLAG from the previous review round (the SduiNode.onView test validating the guard pattern by mirroring the implementation rather than exercising the real component) remains unresolved and is re-raised below.**

### What this PR is doing
**Intent:** Add three additive features to sdui-runtime (Card header/footer slots, one-shot on_view guard, X-Active-Influencer-Id header injection via new useActiveSocialStore) and four state-banner color token pairs to tokens-creator, to unblock the Explore listing surface (brief 264, task 035).
**Delivered:** The diff fully delivers the stated intent. All three sdui-runtime changes are present and correct: Card renderer composes header/items/footer with optional footer_bg_color wrapping; SduiNode.handleViewed gates on firedRef with set-before-dispatch ordering; bff_call injects X-Active-Influencer-Id unconditionally when the store has a value. The tokens-creator changes add all four state-banner bg+text pairs in both light and dark themes with palette.js, palette.d.ts, and palette.test.js updated consistently. Changesets, package.json peer-dep bumps, and sdk-native-sdui 2.8.0 lock-file update are all present. No acceptance items from the linked task are visibly missing.

> ℹ️ 1 previous Zenith review dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- firedRef is set BEFORE dispatch (line 88 before line 90) — the correct ordering to prevent re-entrant double-dispatch is explicitly implemented and the changeset calls it out. This is a subtle correctness detail that is easy to get wrong.
- X-Active-Influencer-Id is unconditionally environment-agnostic (no localhost gate), and the tests explicitly cover both a prod URL and a localhost URL to document and enforce that distinction — good regression coverage for a non-obvious design decision.
- Token pairs are shipped as bg+text couples with AA-contrast rationale documented in the changeset, and the palette.test.js snapshot test is updated to cover all eight new keys — no silent drift surface left open.
- useActiveSocialStore tests clean up after themselves with setActiveInfluencerId(null) in finally blocks and after each test, preventing state leakage across the test suite.

**Read against:**
- `PR body — brief 264 / task 035 acceptance items inferred from PR description`
- `.changeset/runtime-card-onview-active-social.md — stated design intent for all three runtime changes`
- `.changeset/tokens-creator-state-banner-colors.md — stated design intent for token additions`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped PR delivering three additive runtime features and a new token group. The one previously-raised FLAG about the SduiNode.onView test reading raw source via regex/fingerprint is still present and unresolved — the test at SduiNode.onView.test.ts mirrors the guard pattern directly rather than importing and exercising the real component, which is a test-conventions concern worth addressing but not blocking.**

### What this PR is doing
**Intent:** Add three additive runtime features to sdui-runtime (Card header/footer slots, one-shot on_view guard, X-Active-Influencer-Id header injection via new useActiveSocialStore) and four state-banner color token pairs to tokens-creator, unblocking the Explore listing surface.
**Delivered:** The diff fully delivers the stated intent. All three sdui-runtime changes are present and correct: Card renderer composes header/items/footer slots with optional footer_bg_color wrapping (Card.renderer.tsx), SduiNode gains a firedRef one-shot guard (SduiNode.tsx), and bff-call injects X-Active-Influencer-Id unconditionally when the store is non-null (bff-call.ts). The useActiveSocialStore is created, exported from state/index.ts and the package's public index.ts with typed interfaces. The tokens-creator changes add all eight token values (four bg+text pairs) to both theme-light.json and theme-dark.json, palette.js, palette.d.ts, and palette.test.js. Changesets are present for both packages. No acceptance items from the linked task are visibly missing from the diff.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The firedRef is set to true BEFORE dispatch — the set-before-dispatch ordering correctly prevents re-entrant double-dispatch under synchronous callbacks, and the changeset explicitly calls this out.
- useActiveSocialStore is intentionally not localhost-gated (unlike X-Dev-Identity), and this design decision is documented in three places: the store JSDoc, the bff-call.ts comment, and the test file comment block — making the intent clear to future maintainers.
- The footer_bg_color is read from node.config (sibling of data) rather than from the parsed schema value v, with an explanatory comment — correctly handles the schema shape where config is not inside data.
- State cleanup in bff-call tests (finally blocks resetting activeInfluencerId to null) prevents test-order pollution across the suite.
- Both packages have changesets with detailed, accurate descriptions of the changes and their backwards-compatibility guarantees.

**Read against:**
- `PR body — sdui-runtime three-change description`
- `PR body — tokens-creator four state-banner pairs`
- `.changeset/runtime-card-onview-active-social.md`
- `.changeset/tokens-creator-state-banner-colors.md`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/snippets/Card/Card.renderer.tsx [hunk 4]
```diff
@@ -18,18 +33,40 @@ export function CardRenderer(node: Node): React.ReactElement {
...
-        >
-          {v.items?.map((item: Node, i: number) => (
-            <Interpreter key={item.id || i} node={item} />
-          ))}
-        </DSCard>
-      )}
+      {(v) => {
+        // `config` lives as a sibling of `data` on the node itself, not
+        // inside `data`. Reach through the original Node for it.
+        const footerBgColor = (
```
Comment (zenith-oi) [inline packages/sdui-runtime/src/snippets/Card/Card.renderer.tsx]: **FLAG** — [input-validation-at-boundary] The `config.footer_bg_color` field is accessed via a TypeScript type assertion (`node as Node & { config?: { footer_bg_color?: string } }`) rather than through the validated `CardSnippetSchema` parse result. If `CardSnippetSchema` in sdk-native-sdui@2.8.0 already declares `config` on the schema, this cast is redundant and should be replaced with `v.config?.footer_bg_color` (where `v` is the validated parse result). If the schema does NOT declare `config`, then the field is being read from an unvalidated path — a runtime shape mismatch (e.g. `footer_bg_color` arriving as a non-string) would pass TypeScript silently and reach `<Box bg={...}>` unchecked. Either thread `config` through the Zod schema and read it from `v`, or add a narrow Zod parse at this call site. The comment on L37-38 acknowledges the reach-through but doesn't address the validation gap.

**Review by zenith-oi — DISMISSED** [review]
## Zenith Code Review

**Clean, well-scoped PR that delivers three additive runtime features and four token pairs. Code is correct, tests are present and meaningful, changesets are accurate. One minor concern worth the engineer's attention: the `config` field is reached via an unsafe cast on the raw Node rather than being part of the validated schema, which could silently pass wrong types at runtime.**

### What this PR is doing
**Intent:** Add three additive runtime features to sdui-runtime (Card header/footer slots, one-shot on_view guard, X-Active-Influencer-Id header injection via new useActiveSocialStore) and four state-banner color token pairs to tokens-creator, to unblock the Explore listing surface (brief 264, task 035).
**Delivered:** The diff fully delivers all three runtime changes and the token additions as described. Card renderer now composes header/items/footer slots with optional footer background color. SduiNode.handleViewed is guarded by a per-instance firedRef. useActiveSocialStore is created, exported, and threaded into bff_call. Four state-banner bg+text token pairs are added to both light and dark themes with palette.js and palette.d.ts updated. Tests cover all three runtime changes and the new palette group. No acceptance items from the linked task are visibly missing from the diff. The one pre-existing test failure noted in the PR body is unrelated to this PR.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The firedRef set-before-dispatch ordering is correct and the regression test explicitly documents the re-entrant synchronous callback edge case — this is exactly the kind of subtle ordering bug that causes duplicate analytics in production and the test comment makes the intent clear.
- The bff_call tests cover all three cases (prod URL, localhost URL, null store) and explicitly document why X-Active-Influencer-Id is not localhost-gated unlike X-Dev-Identity — good institutional knowledge preserved in test comments.
- State cleanup in bff_call tests (finally blocks resetting the store to null) prevents test-order pollution — correct pattern for shared Zustand store state in node:test suites.
- Token pairs are well-reasoned: the changeset explains why urgent-orange is distinct from action-required amber and from negative red, and the dark-theme 18%-opacity approach is a principled choice that keeps banners recessive rather than competing with body content.

**Read against:**
- `PR body — brief 264 wiki task 035 (tasks/035-bff-campaigns-explore-listing.md)`
- `.changeset/runtime-card-onview-active-social.md — stated design intent for all three runtime changes`
- `.changeset/tokens-creator-state-banner-colors.md — stated design intent for token additions`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

**Clean, well-scoped PR delivering three additive runtime features and four token pairs. Code is correct, tests are present and meaningful, and the changeset descriptions are thorough. No blocking issues found.**

### What this PR is doing
**Intent:** Add three additive features to sdui-runtime (Card header/footer/items slot rendering, one-shot on_view guard, X-Active-Influencer-Id header injection via new useActiveSocialStore) and four state-banner color token pairs to tokens-creator, to unblock the Explore listing surface (brief 264, task 035).
**Delivered:** The diff fully delivers all three sdui-runtime changes and the tokens-creator additions as described. Card renderer correctly composes header/items/footer with optional footer background color. The one-shot firedRef guard is correctly placed before dispatch (set-before-dispatch ordering). useActiveSocialStore is a clean Zustand store exported on the public surface. bff-call injects X-Active-Influencer-Id unconditionally (not localhost-gated). Token pairs are present in both light and dark theme JSON files with matching palette.js and palette.d.ts entries. Tests cover all three new behaviors. No linked task acceptance items are available to check against individually, but the PR body's stated deliverables are all present in the diff.

No issues found.

**Positive:**
- The firedRef set-before-dispatch ordering is explicitly called out in both the changeset and the PR body, and the regression test directly documents the two failure modes it guards against (gate removed, gate placed after dispatch) — this is exactly the right level of test commentary for a subtle correctness invariant.
- The config safeParse in CardRenderer correctly avoids an `as` cast on the raw node's config field, dropping malformed footer_bg_color values rather than letting them reach the renderer — good defensive boundary discipline.
- useActiveSocialStore tests clean up after themselves with setActiveInfluencerId(null) in finally blocks and explicit teardown, preventing state leakage across test files in the shared Zustand singleton.
- The distinction between X-Active-Influencer-Id (all environments) and X-Dev-Identity (localhost-only) is clearly documented in three places: the store JSDoc, the bff-call comment, and the test comment block — future maintainers won't accidentally localhost-gate the production header.

**Read against:**
- `PR body — sdui-runtime: Card renderer header/footer slots, on_view one-shot, X-Active-Influencer-Id header`
- `PR body — tokens-creator: state-banner color token pairs (neutral/action-required/success/urgent)`
- `.changeset/runtime-card-onview-active-social.md — changeset description as design intent`
- `.changeset/tokens-creator-state-banner-colors.md — changeset description as design intent`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*

File: packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts (hunk 3 above)
Comment (zenith-oi) [inline packages/sdui-runtime/src/sdui-node/__tests__/SduiNode.onView.test.ts:30]: **FLAG** — [test-conventions] The test re-implements the one-shot guard inline (a local `firedRef` + `handleViewed` closure) rather than importing and exercising the real `SduiNode` component. This means the test does not actually catch a regression where the guard is removed or misplaced inside `SduiNode.tsx` — it only validates the abstract pattern. If `firedRef.current = true` is moved to after the dispatch in the real component, this test still passes. A test that exercises the real component (even via a lightweight React test renderer or by calling the exported `handleViewed` callback directly if it were exposed) would catch that regression. Consider either: (a) using `@testing-library/react-native` to mount `SduiNode` with a mock `Viewable` and assert dispatch count, or (b) at minimum adding a comment acknowledging this is a pattern-level smoke test, not a component-level regression test, so future maintainers understand the coverage gap.

**Review by zenith-oi — APPROVED** [review]
## Zenith Code Review

> ⚠️ **Round 4 — engineer attention requested.** The reviewer-engineer loop has reached round 4; this is the threshold where auto-iteration typically stops being productive (either the review is missing context the engineer has, or there's a genuine design call to make). Auto-fixes are still posted, but please review this thread end-to-end rather than relying on another round to converge.

**Clean, well-scoped PR delivering three additive runtime features and eight new design tokens. Code is correct, tests cover the new behaviour, and the changeset descriptions are accurate. One pre-existing FLAG from the previous round (test-conventions on SduiNode.onView.test.ts) remains unaddressed — the test mirrors the guard inline rather than importing the real component — but this is a low-severity concern and the engineer has not changed that file in this push.**

### What this PR is doing
**Intent:** Add three additive features to sdui-runtime (Card header/footer slots, one-shot on_view guard, X-Active-Influencer-Id header injection via new useActiveSocialStore) and four state-banner color token pairs to tokens-creator, to unblock the Explore listing surface (brief 264, task 035).
**Delivered:** The diff fully delivers the stated intent. All three sdui-runtime changes are present and correct: Card renderer composes header/items/footer with optional footer_bg_color wrapping (Card.renderer.tsx), SduiNode gains a per-instance firedRef one-shot guard (SduiNode.tsx), and bff-call injects X-Active-Influencer-Id unconditionally when the store has a value (bff-call.ts + useActiveSocialStore.ts). The tokens-creator changes add all eight token values (four bg/text pairs) in both light and dark themes, with palette.js, palette.d.ts, and palette.test.js all updated consistently. Changesets are present for both packages. No acceptance items from the linked task are visible in the diff context, but the PR body's stated deliverables are all present.

> ℹ️ 2 previous Zenith reviews dismissed. Inline comments from those reviews are superseded — only comments from **this review** are current.

**1 FLAG** — advisory, not merge-blocking. See inline comments.

**Positive:**
- The set-before-dispatch ordering in the firedRef guard (line 88 sets true before line 90 dispatches) is correct and the changeset explicitly calls this out — shows the author thought carefully about re-entrant callback edge cases.
- The config safeParse in Card.renderer.tsx (lines 43-48) correctly avoids casting the raw node and drops a malformed footer_bg_color rather than letting it reach the renderer — good defensive boundary discipline.
- The X-Active-Influencer-Id header is intentionally NOT localhost-gated (unlike X-Dev-Identity), and this distinction is documented in three places (source comment, test comment, changeset) — makes the design decision auditable.
- Test cleanup is disciplined: every bff-call test that sets activeInfluencerId resets it to null in a finally block, preventing state leakage across test files.
- Token pairs are complete and symmetric: every new token has both a light and dark theme value, and the palette.d.ts, palette.js, and palette.test.js are all updated in lockstep.

**Read against:**
- `PR body — sdui-runtime: Card renderer header/footer slots, on_view one-shot, X-Active-Influencer-Id header`
- `PR body — tokens-creator: state-banner color tokens (neutral/action-required/success/urgent bg+text pairs)`
- `.changeset/runtime-card-onview-active-social.md — changeset description matches implementation`
- `.changeset/tokens-creator-state-banner-colors.md — changeset description matches token additions`

---
*Generated by [Zenith AI CTO](https://github.com/one-impression/zenith-coding-agent)*
